otodom/task_1/mk/otodom.py

```diff
@@ -0,0 +1,165 @@
+import json
+import re
+import sys
+
+import requests
+from bs4 import BeautifulSoup
+
+HEADERS = {
+    "Access-Control-Allow-Origin": "*",
+    "Content-Type": "text/html; charset=utf-8",
+    "Accept-Language": "pl-PL,pl;q=0.9,en-US;q=0.8,en;q=0.7",
+    "Sec-Ch-Ua": '"Chromium";v="118", "Google Chrome";v="118", "Not=A?Brand";v="99"',
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36",  # noqa: E501
+}
+
+
+class Crawler:
+    def getPage(self, url):
+        try:
+            req = requests.get(url, headers=HEADERS)
+        except requests.exceptions.RequestException:
+            print(f"Failed to retrieve the webpage. Status code: {req.status_code}")
+            return None
+        return BeautifulSoup(req.text, "html.parser")
+
+    def getLinks(self, url):
+        bs = self.getPage(url)
+        if bs is not None:
+            promoted_listings = bs.find(
+                "div", {"data-cy": "search.listing.promoted"}
+            ).find_all("a", {"class": "css-1hfdwlm e1dfeild2"})
```

Comment:
same as in line 104 try to find more general approach, where attributes are defined by a human.